crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -141,38 +145,198 @@ impl Persistence {
+                let token = eth::H160(price.token.0).into();
+                let price = big_decimal_to_u256(&price.price)
+                    .ok_or(domain::auction::InvalidPrice)
+                    .and_then(|p| domain::auction::Price::new(p.into()))
+                    .map_err(error::Auction::Price)?;
+                prices.insert(token, price);
+            }
+            prices
+        };
+
+        let (fee_policies, missing_orders) = {
+            let mut missing_orders = HashSet::new();
+            let mut fee_policies = HashMap::new();
+            for order in settlement.order_uids().cloned() {
+                match database::orders::read_order(&mut ex, &ByteArray(order.0))
+                    .await
+                    .context("fetch order")?
+                {
+                    Some(_) => {
+                        let quote = database::orders::read_quote(&mut ex, &ByteArray(order.0))
+                            .await
+                            .context("fetch quote")?
+                            .map(dto::quote::into_domain)
+                            .transpose()
+                            .map_err(|_| error::Auction::DbConversion("quote overflow"))?
+                            .ok_or(error::Auction::MissingQuote)?;
+
+                        let policies =
+                            database::fee_policies::fetch(&mut ex, auction, ByteArray(order.0))
+                                .await
+                                .context("fetch fee policies")?;
+
+                        for policy in policies {
+                            let policy = dto::fee_policy::into_domain(policy, &quote)?;
+                            fee_policies
+                                .entry(order)
+                                .or_insert_with(Vec::new)
+                                .push(policy);
+                        }
+                    }
+                    None => {
+                        missing_orders.insert(order);
+                    }
+                }
+            }
+            (fee_policies, missing_orders)
+        };
+
+        let solution = {
+            // TODO: stabilize the solver competition table to get promised solution.
+            // https://github.com/cowprotocol/services/issues/2716
+            // for now, this is a `good enough` solution to unblock other tasks.
+            let solver_competition = database::solver_competition::load_by_id(&mut ex, auction)
+                .await
+                .context("load solver competition")?
+                .ok_or(error::Auction::SolverCompetition(
+                    "missing solver competition",
+                ))?;
+            let competition: model::solver_competition::SolverCompetitionDB =
+                serde_json::from_value(solver_competition.json)
+                    .map_err(|_| error::Auction::SolverCompetition("json conversion"))?;
+            let winning_solution = competition.solutions.last().unwrap();
```

Comment:
There is already error handling all over the place. I think there is no need to `.unwrap()` here then.